gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/GenericStatement.java

```diff
@@ -133,6 +133,8 @@ Licensed to the Apache Software Foundation (ASF) under one or more
 
         public static final Pattern SKIP_CANCEL_STMTS = Pattern.compile(
             "^\\s*(DROP|TRUNCATE)\\s+(TABLE|INDEX)\\s+", Pattern.CASE_INSENSITIVE);
+	public static final Pattern DELETE_STMT = Pattern.compile(
+			"^\\s*DELETE\\s+FROM\\s+.*", Pattern.CASE_INSENSITIVE);
         private static final Pattern ignoreStmts = Pattern.compile(
```

Comment:
Ok, so perhaps we can do regex check for the case where it is before prepMinion, but then use the parse tree to figure out precisely. So being a bit conservative in the regex check should be fine (in allowing go ahead), then do a more proper check using the QT.